File: PHP/src/OneLang/One/Transforms/ConvertToMethodCall.php
<?php

namespace OneLang\One\Transforms\ConvertToMethodCall;

use OneLang\One\Ast\Expressions\Expression;
use OneLang\One\Ast\Expressions\ElementAccessExpression;
use OneLang\One\Ast\Expressions\UnresolvedCallExpression;
use OneLang\One\Ast\Expressions\PropertyAccessExpression;
use OneLang\One\Ast\Expressions\BinaryExpression;
use OneLang\One\Ast\Expressions\StringLiteral;
use OneLang\One\AstTransformer\AstTransformer;

class ConvertToMethodCall extends AstTransformer {
    function __construct() {
        parent::__construct("ConvertToMethodCall");
        
    }
    
    protected function visitExpression($expr) {
        $origExpr = $expr;
        
        $expr = parent::visitExpression($expr);
        
        if ($expr instanceof BinaryExpression && $expr->operator === "in")
            $expr = new UnresolvedCallExpression(new PropertyAccessExpression($expr->right, "hasKey"), array(), array($expr->left));
        
        $expr->parentNode = $origExpr->parentNode;
        return $expr;
    }
}

File: Python/OneLang/One/Transforms/ConvertToMethodCall.py
from onelang_core import *
import OneLang.One.Ast.Expressions as exprs
import OneLang.One.AstTransformer as astTrans

class ConvertToMethodCall(astTrans.AstTransformer):
    def __init__(self):
        super().__init__("ConvertToMethodCall")
    
    def visit_expression(self, expr):
        orig_expr = expr
        
        expr = super().visit_expression(expr)
        
        if isinstance(expr, exprs.BinaryExpression) and expr.operator == "in":
            expr = exprs.UnresolvedCallExpression(exprs.PropertyAccessExpression(expr.right, "hasKey"), [], [expr.left])
        
        expr.parent_node = orig_expr.parent_node
        return expr